cm_service_manager.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import sys
+import smtplib
+import cm_api
+from cm_api.api_client import ApiResource
+
+parser = argparse.ArgumentParser(description='Restart a CM cluster service')
+
+parser.add_argument('-m', '--cm_host', required=True,
+                    help="hostname / fqdn of the CM UI ('cloudera.domain.com')")
+parser.add_argument('-u', '--username', default=os.environ['USER'],
+                    help="username for the CM API authentication ('johndoe')")
+parser.add_argument('-p', '--password', required=True,
+                    help="password for the CM API authentication ('Pa55w0rd')")
+parser.add_argument('-l', '--cluster', required=False,
+                    help="cluster to restart service in ('mycluster')")
+parser.add_argument('-s', '--service', required=False,
+                    help="service to restart ('hdfs')")
+parser.add_argument('-e', '--smtp_server', required=False,
+                    help="smtp server to send e-mail notifications")
+
+args = parser.parse_args()
+print args
+
+# https://cloudera.github.io/cm_api/docs/quick-start/
+api = ApiResource(args.cm_host, username=args.username, password=args.password)
+
+def select_from_items(item_type, items):
+    print("\n".join(["Choose a %s" % item_type] + [i.lower() for i in items]))
+    item = raw_input().lower()
+    print "Input received: %s" % item
+    if item in items:
+        return item
+    else:
+        print 'Bad input received...'
+        return select_from_items(item_type, items)
+
+
+# Rolling restart method (using few defaults - Role types, role names. It will take all)
+def rolling_restart(service_rr):
+    print service_rr
+    try:
+        #if service_rr.name.lower() in ['hive', 'hue', 'oozie', 'spark']:
+        #    raise AttributeError("Rolling Restart is not available for '%s' " % service.name.lower())
+        batch_size = raw_input("Number of worker roles to restart together in a batch \n")
+        fail_count_threshold = raw_input("Stop rolling restart if more than this number of worker batches fail \n")
+
+        options = ["true", "false"]    # options for stale config
+        stale_config_only = select_from_items('option to Restart roles on nodes with stale config? Default is false \n', options)   # Needs to be a user selection
+
+        rr_command = service_rr.rolling_restart(batch_size, fail_count_threshold, None, stale_config_only, None, None, None).wait()
+    except:
+        print "WARNING: Rolling Restart is not available for '%s' " % service.name.lower()
+        print "Executing Restart on service '%s' " % service.name.lower()
+        rr_command = service.restart().wait()
+    return rr_command
```

Comment:
Added the try block.
But one thing that I think is messing up would be : when service = 'hive/oozie/hue', then the batch_size etc would still be asked to input from the user, which would be unnecessary and which I do not want. 
So i was thinking , inside of try block, can a single 'if' statement be used to verify the service and raise exception if service is one those? (I have commented out in line 46-47). This way if the service doesn't support RR, then user would not need to unnecessarily input those details and except block will be executed.